src/avram/type.cr

```diff
@@ -32,5 +42,8 @@ module Avram::Type
   end
 
   class FailedCast
+    def value
```

Comment:
Why does `FailedCast` need a value? Could you link me to the code that needs it? Just making sure I understand :)

Reply:
this isn't totally necessary, but without it there's a compilation error. If I remove it, then the `parse!` method above on line 24 would have to add in a catch for when `parse(value)` doesn't return a `SuccessfulCast`. Alternatively, the array parse method on line 20 above could just not return a `FailedCast`... but that adds in some other weird stuff. I'm down for either way though if you have a better suggestion that makes it cleaner.